File: src/main/java/ru/stnk/RestTestAPI/configuration/SessionConfig.java
package ru.stnk.RestTestAPI.configuration;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.security.core.session.SessionRegistry;
import org.springframework.session.jdbc.config.annotation.web.http.EnableJdbcHttpSession;
import org.springframework.session.security.web.authentication.SpringSessionRememberMeServices;
import org.springframework.session.web.context.AbstractHttpSessionApplicationInitializer;
import org.springframework.session.web.http.CookieHttpSessionIdResolver;
import org.springframework.session.web.http.DefaultCookieSerializer;
import org.springframework.session.web.http.HttpSessionIdResolver;

@Configuration
//Из-за включенной аннотации не создаются таблицы Spring_Session
//@EnableJdbcHttpSession
public class SessionConfig extends AbstractHttpSessionApplicationInitializer {
    /*@Bean
    public HttpSessionIdResolver httpSessionIdResolver() {
        return HeaderHttpSessionIdResolver.xAuthToken();
    }*/

    @Bean
    public HttpSessionIdResolver httpSessionIdResolver() {
        CookieHttpSessionIdResolver resolver = new CookieHttpSessionIdResolver();
        DefaultCookieSerializer cookieSerializer = new DefaultCookieSerializer();
        cookieSerializer.setUseBase64Encoding(false);
        cookieSerializer.setUseHttpOnlyCookie(false);
        resolver.setCookieSerializer(cookieSerializer);
        return resolver;
    }

    @Bean
    public SpringSessionRememberMeServices rememberMeServices() {
        SpringSessionRememberMeServices rememberMeServices = new SpringSessionRememberMeServices();
        rememberMeServices.setAlwaysRemember(true);
        return rememberMeServices;
    }

}

File: src/main/kotlin/ru/stnk/resttestapi/configuration/SessionConfig.kt
package ru.stnk.resttestapi.configuration

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.session.security.web.authentication.SpringSessionRememberMeServices
import org.springframework.session.web.context.AbstractHttpSessionApplicationInitializer
import org.springframework.session.web.http.CookieHttpSessionIdResolver
import org.springframework.session.web.http.DefaultCookieSerializer
import org.springframework.session.web.http.HttpSessionIdResolver

@Configuration
//Из-за включенной аннотации не создаются таблицы Spring_Session
//@EnableJdbcHttpSession
class SessionConfig : AbstractHttpSessionApplicationInitializer() {
    /*@Bean
    public HttpSessionIdResolver httpSessionIdResolver() {
        return HeaderHttpSessionIdResolver.xAuthToken();
    }*/

    @Bean
    fun httpSessionIdResolver(): HttpSessionIdResolver {
        val resolver = CookieHttpSessionIdResolver()
        val cookieSerializer = DefaultCookieSerializer()
        cookieSerializer.setUseBase64Encoding(false)
        cookieSerializer.setUseHttpOnlyCookie(false)
        resolver.setCookieSerializer(cookieSerializer)
        return resolver
    }

    @Bean
    fun rememberMeServices(): SpringSessionRememberMeServices {
        val rememberMeServices = SpringSessionRememberMeServices()
        rememberMeServices.setAlwaysRemember(true)
        return rememberMeServices
    }

}
